### PR TITLE
Fix reshape on scalars/empty arrays

### LINKED ISSUE #157: Reshape on empty arrays
The following case doesn't work in `pytato` but `numpy` happily reshapes it to anything.

```python
x = pt.make_placeholder("x", shape=(0,), dtype=np.float64)
assert pt.reshape(x, (128, 0, 17)).shape == (128, 0, 17)
```

and 

```python
x = pt.make_placeholder("x", shape=(), dtype=np.float64)
pt.reshape(x, (1,))
```
seems to fail at code generation in [here](https://github.com/inducer/pytato/blob/7fcf1a65be0873f9eea991db74713e64685bf9f9/pytato/codegen.py#L411).

xref: inducer/arraycontext#91

### PR DESCRIPTION
**Pull request by kaushikcfd**
Closes #157

### CODE REVIEW

File: pytato/array.py [hunk 1]
```diff
@@ -1844,7 +1847,7 @@ def reshape(array: Array, newshape: Union[int, Sequence[int]],
         if not isinstance(new_axislen, int):
             raise ValueError("Symbolic reshapes not allowed.")
 
-        if not(new_axislen > 0 or new_axislen == -1):
+        if not(new_axislen >= 0 or new_axislen == -1):
             raise ValueError("newshape should be either sequence of positive ints or"
```
Comment (alexfikl) [inline pytato/array.py:1851]:
```suggestion
            raise ValueError("newshape should be either sequence of non-negative ints or"
```
?

**Review by alexfikl — COMMENTED** [review]
Looks good to me! Thanks for taking care of it so quickly!

File: pytato/array.py [hunk 2]
```diff
@@ -572,6 +572,8 @@ def imag(self) -> ArrayOrScalar:
 
     def reshape(self, *shape: Union[int, Sequence[int]], order: str = "C") -> Array:
         import pytato as pt
+        if len(shape) == 0:
+            raise ValueError("reshape takes at least one argument (0 given)")
```
Comment (inducer) [inline pytato/array.py:576]:
```suggestion
            raise TypeError("reshape takes at least one argument (0 given)")
```

File: pytato/array.py [hunk 3]
```diff
@@ -1844,9 +1849,9 @@ def reshape(array: Array, newshape: Union[int, Sequence[int]],
         if not isinstance(new_axislen, int):
             raise ValueError("Symbolic reshapes not allowed.")
 
-        if not(new_axislen > 0 or new_axislen == -1):
-            raise ValueError("newshape should be either sequence of positive ints or"
-                    " -1")
+        if new_axislen < -1:
+            raise ValueError("newshape should be either sequence of non-negative"
```
Comment (inducer) [inline pytato/array.py:1853]:
```suggestion
            raise ValueError("newshape should be either a sequence of non-negative"
```

**Review by inducer — COMMENTED** [review]
Thanks, LGTM! I'll apply these two minor things myself.

**Pull request merged**